src/Microsoft.Health.Fhir.Core/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Health.Fhir.Core.Extensions;
+
+public static class ServiceCollectionExtensions
+{
+    public static IServiceCollection RemoveServiceTypeExact<TConcreteClass, TServiceType>(this IServiceCollection serviceCollection)
```

Comment:
Why is this check necessary for the notification handlers? This looks like it prevents duplicate registrations, but I don't see why this is a special concern for the notification handlers.

Reply:
The problem stems from the behavior that Mediatr registers all Handlers as Transient. If we want to change these to be Scoped or Singleton we need to remove the original registration. We should be able to register multiple Handlers for a Notification, this is part of what makes them powerful